src/wallet3/default_daemon_comms.cpp

```diff
@@ -443,10 +467,50 @@ namespace wallet
     return fut;
   }
 
+  std::future<std::string>
+  DefaultDaemonComms::ons_names_to_owners(const std::string& name_hash, const uint16_t type)
+  {
+    auto p = std::make_shared<std::promise<std::string> >();
+    auto fut = p->get_future();
+    auto req_cb = [p=std::move(p)](bool ok, std::vector<std::string> response)
+    {
+
+      oxenc::bt_dict_consumer dc{response[1]};
+
+      if (not dc.skip_until("result"))
+      {
+        auto reason = dc.consume_string();
+        p->set_value(std::string("ONS names to owners rejected, reason: ") + reason);
+        return;
+      }
+      auto result_list = dc.consume_list_consumer();
+      const auto result = result_list.consume_dict_data();
+
+      p->set_value(std::string(result));
```

Comment:
This is returning a (delayed) serialized value, which doesn't feel right: the caller right now has to know about how to deserialize the result, but knows nothing about serializing the input.  I think instead this should be deserializing the result into a pair or a vector or something so that you have simple values in -> simple values out.